CMakeLists.txt

```diff
@@ -110,12 +113,6 @@ option_with_flags(ENABLE_CODE_COVERAGE "Enables details on code coverage" OFF
                   "-ftest-coverage")
 option_with_flags(ENABLE_BOUNDS_CHECK "Enables bounds check in Fortran" OFF
                   "-ftrapuv -check all -fpstkchk" "-fcheck=all" "-fbounds-check -fcheck-array-temporaries")
-option_with_flags(ENABLE_ASAN "Enables address sanitizer" OFF
```

Comment:
I know only recent Clang and GCC support these flags, but why skip the `check_cxx_compiler_flag` in `option_with_flags`?

Reply:
On my system the flags are not passed to the linker when using `check_cxx_compiler_flag` which are needed and the check fails.

Reply:
try_compile doesn't like it. But if it could be made to work, that'd def. be better.

Reply:
`try_compile` or `check_cxx_source_compiles` would probably be more flexible/robust in this situation. I'll try it out.

Reply:
The following works:
```
cmake_minimum_required(VERSION 3.1 FATAL_ERROR)

project(recipe-01 CXX)

include(CheckCXXSourceCompiles)
set(_source "int main() { return 0; }")
set(CMAKE_REQUIRED_FLAGS "-fsanitize=address -fno-omit-frame-pointer")
check_cxx_source_compiles("${_source}" _asan_works)
```
 `check_cxx_source_compiles` forwards the flags to the linker too.

Reply:
interesting! maybe `check_cxx_source_compiles` should replace the fn in cmake/psi4OptionsTools? (though that's in a lot of codes now and shouldn't be done casually.)